test/Router.js

```diff
@@ -839,6 +839,122 @@ describe('Router', function() {
 			});
 	});
 
+	it('should prevent navigation when promise returned by beforeActivateHandler resolves to true', function(done) {
+		router = new Router({
+			beforeActivateHandler: () => {
+				return new CancellablePromise(resolve => {
+					setTimeout(() => {
+						resolve(true);
+					}, 0);
+				});
+			},
+			path: '/path',
+			component: CustomComponent,
+		});
+
+		const {pathname} = window.location;
+
+		Router.router()
+			.navigate('/path')
+			.catch(err => {
+				assert.equal(Router.getActiveComponent(), null);
+				assert.equal(err.message, 'Cancelled by next screen');
+				assert.equal(pathname, window.location.pathname);
+
+				done();
+			});
+	});
+
+	it('should not prevent navigation when promise returned by beforeActivateHandler resolves to false', function(done) {
+		router = new Router({
+			beforeActivateHandler: () => {
+				return new CancellablePromise(resolve => {
+					setTimeout(() => {
+						resolve(false);
+					}, 0);
+				});
+			},
+			path: '/path',
+			component: CustomComponent,
+		});
+
+		Router.router()
+			.navigate('/path')
+			.then(() => {
+				assert.equal('/path', window.location.pathname);
+				assert.ok(Router.getActiveComponent() instanceof CustomComponent);
+
+				done();
+			});
+	});
+
+	it('should prevent navigation when beforeActivateHandler returns true', function(done) {
+		router = new Router({
+			beforeActivateHandler: () => {
+				return true;
+			},
+			path: '/path',
+			component: CustomComponent,
+		});
+
+		const {pathname} = window.location;
+
+		Router.router()
+			.navigate('/path')
+			.catch(err => {
+				assert.equal(Router.getActiveComponent(), null);
+				assert.equal(err.message, 'Cancelled by next screen');
+				assert.equal(pathname, window.location.pathname);
+
+				done();
+			});
+	});
+
+	it('should not prevent navigation when beforeActivateHandler returns false', function(done) {
+		router = new Router({
+			beforeActivateHandler: () => {
+				return false;
+			},
+			path: '/path',
+			component: CustomComponent,
+		});
+
+		Router.router()
+			.navigate('/path')
+			.then(() => {
+				assert.equal('/path', window.location.pathname);
+				assert.ok(Router.getActiveComponent() instanceof CustomComponent);
+
+				done();
+			});
+	});
+
+	it('should prevent navigation if beforeActivateHandler given by name returns "true"', function(done) {
```

Comment:
Are we missing this test for `beforeDeactivateHandler`?

Reply:
There was a test for this behavior prior to any of these changes, the name of the test is just a little different. https://github.com/metal/metal-router/pull/34/files/a1a749a82e265bd4fee2ed1d8f5dd24d0ab5f19d#diff-2fa50a416f60f046085d3bd37499a551R1014

Reply:
Got it!